sqlglot/dialects/dialect.py

```diff
@@ -993,3 +993,33 @@ def no_last_day_sql(self: Generator, expression: exp.LastDay) -> str:
     minus_one_day = exp.func("date_sub", plus_one_month, 1, "day")
 
     return self.sql(exp.cast(minus_one_day, "date"))
+
+
+def remove_target_from_merge(self: Generator, expression: exp.Merge) -> str:
```

Comment:
this should go in transforms.py

Reply:
ah i see

Reply:
why do we need transforms.preprocess at all then? just call transform

Reply:
Ah, sure. Perhaps we can also get rid of the closure and just do this in a single nesting level, deferring to `Generator.merge_sql` (which will also be reached if we call `self.sql(merge_expression)` at the end of this method

Reply:
@eakmanrq I can try out this refactor in a followup PR if you want

Reply:
sounds good I will just merge this as is